apps/api/src/app/events/usecases/resume-delayed/resume-delayed.usecase.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { JobStatusEnum, JobRepository } from '@novu/dal';
+import { isBefore } from 'date-fns';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { ResumeDelayedCommand } from './resume-delayed.command';
+
+@Injectable()
+export class ResumeDelayed {
+  constructor(private jobRepository: JobRepository) {}
+
+  public async execute(command: ResumeDelayedCommand): Promise<void> {
+    const job = await this.jobRepository.findOne({
+      _environmentId: command.environmentId,
+      transactionId: command.transactionId,
+      status: JobStatusEnum.CANCELED,
+    });
+
+    if (!job) {
+      throw new NotFoundException(`Job with transactionId ${command.transactionId} was not found`);
+    }
+
+    if (isBefore(new Date(job?.expireAt as string), new Date())) {
```

Comment:
expireAt is the ttl value, its the date when the entity will be archived. It is unrelated to the delay amount. 
You need to check if the calculation of the delay did not pass